fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/deployment/auth/config/JWTAuthContextInfoProvider.java

```diff
@@ -52,36 +58,82 @@
         values to "NONE" as Optional Strings are populated with a ConfigProperty.defaultValue if they are absent. Fix this when MP-Config
         is repaired.
          */
-        if (NONE.equals(publicKeyPemEnc.get()) && NONE.equals(jwksUri.get())) {
+        if (NONE.equals(publicKeyPemEnc.get()) && NONE.equals(jwksUri.get()) &&
+                NONE.equals(super.getMpJwtublicKey().get()) && NONE.equals(super.getMpJwtLocation().get())) {
             return Optional.empty();
         }
         JWTAuthContextInfo contextInfo = new JWTAuthContextInfo();
-        if (publicKeyPemEnc.isPresent() && !NONE.equals(publicKeyPemEnc.get())) {
+        // Look to MP-JWT values first
+        if (super.getMpJwtublicKey().isPresent() && !NONE.equals(super.getMpJwtublicKey().get())) {
+            // Need to decode what this is...
+            Optional<String> mpJwtublicKey = super.getMpJwtublicKey();
+            try {
+                RSAPublicKey pk = (RSAPublicKey) KeyUtils.decodeJWKSPublicKey(mpJwtublicKey.get());
+                contextInfo.setSignerKey(pk);
+                log.debugf("mpJwtublicKey parsed as JWK(S)");
+            } catch (Exception e) {
+                // Try as PEM key value
+                log.debugf("mpJwtublicKey failed as JWK(S), %s", e.getMessage());
+                try {
+                    RSAPublicKey pk = (RSAPublicKey) KeyUtils.decodePublicKey(mpJwtublicKey.get());
+                    contextInfo.setSignerKey(pk);
+                    log.debugf("mpJwtublicKey parsed as PEM");
+                } catch (Exception e1) {
+                    throw new DeploymentException(e1);
+                }
+            }
+        } else if (publicKeyPemEnc.isPresent() && !NONE.equals(publicKeyPemEnc.get())) {
             try {
                 RSAPublicKey pk = (RSAPublicKey) KeyUtils.decodePublicKey(publicKeyPemEnc.get());
                 contextInfo.setSignerKey(pk);
             } catch (Exception e) {
                 throw new DeploymentException(e);
             }
         }
-        if (issuedBy != null && !issuedBy.equals(NONE)) {
+
+        String mpJwtIssuer = super.getMpJwtIssuer();
+        if (mpJwtIssuer != null && !mpJwtIssuer.equals(NONE)) {
+            contextInfo.setIssuedBy(mpJwtIssuer);
+        } else if (issuedBy != null && !issuedBy.equals(NONE)) {
             contextInfo.setIssuedBy(issuedBy);
+        } else {
+            // If there is no expected issuer configured, don't validate it; new in MP-JWT 1.1
+            contextInfo.setRequireIssuer(false);
         }
+
+        Optional<Boolean> mpJwtRequireIss = super.getMpJwtRequireIss();
+        if (mpJwtRequireIss != null && mpJwtRequireIss.isPresent()) {
+            contextInfo.setRequireIssuer(mpJwtRequireIss.get());
+        } else {
+            // Default is to require iss claim
+            contextInfo.setRequireIssuer(true);
```

Comment:
A couple lines above, we have

```java
// If there is no expected issuer configured, don't validate it; new in MP-JWT 1.1
contextInfo.setRequireIssuer(false);
```

And here, we do `contextInfo.setRequireIssuer(true);`. That looks strange to me.